### PR TITLE
fix(channel): wrong case clause when alias is inexistent

### LINKED ISSUE #6978: Crash when publishing to topic alias which doesn't exist
### Subject of the issue
EMQ X 4.3.11 crashes the connection when attempting to publish to a non-existent topic alias.

### Your environment

For EMQ X 4.3 or newer, please provide the log archive generated by [node_dump tool](https://github.com/emqx/emqx/blob/master/bin/node_dump) 

- EMQ X version (e.g. `emqx_ctl status`): 4.3.11 Docker image
- Hardware configuration (e.g. `lscpu`): 
- OS (e.g. `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Erlang/OTP version (in case you build emqx from source code): 
- Others:

### Steps to reproduce
Using the Paho MQTT 5 tester (https://github.com/eclipse/paho.mqtt.testing/blob/master/interoperability/client_test5.py#L765-L771), publishing to a topic alias (1) which hasn't been created will cause a crash instead of a disconnect packet.

### Expected behaviour
A disconnect packet should be sent with a proper reason code.

### Actual behaviour
Crash with:
```
2022-02-10T18:32:54.851450+00:00 [error] myclientid@172.17.0.1:58334 crasher: initial call: emqx_connection:init/4, pid: <0.2066.0>, registered_name: [], error: {{case_clause,error},[{emqx_channel,process_alias,2,[{file,"emqx_channel.erl"},{line,1332}]},{emqx_misc,pipeline,3,[{file,"emqx_misc.erl"},{line,140}]},{emqx_channel,process_publish,2,[{file,"emqx_channel.erl"},{line,516}]},{emqx_connection,with_channel,3,[{file,"emqx_connection.erl"},{line,664}]},{emqx_connection,process_msg,2,[{file,"emqx_connection.erl"},{line,364}]},{emqx_connection,process_msg,2,[{file,"emqx_connection.erl"},{line,370}]},{emqx_connection,handle_recv,3,[{file,"emqx_connection.erl"},{line,328}]},{proc_lib,wake_up,3,[{file,"proc_lib.erl"},{line,236}]}]}, ancestors: [<0.1768.0>,<0.1767.0>,esockd_sup,<0.1557.0>], message_queue_len: 0, messages: [], links: [<0.1768.0>], dictionary: [{recv_pkt,4},{outgoing_bytes,53},{guid,{1644517974345319,74977244088338,1}},{outgoing_pubs,1},{incoming_bytes,78},{acl_keys_q,{[{publish,<<"TopicA">>}],[]}},{send_msg,1},{acl_cache_size,1},{send_pkt,3},{'$logger_metadata$',#{clientid => "myclientid",peername => "172.17.0.1:58334"}},{recv_msg,2},{{publish,<<"TopicA">>},{allow,1644517974345}},{incoming_pubs,2}], trap_exit: false, status: running, heap_size: 2586, stack_size: 28, reductions: 11671; neighbours:
2022-02-10T18:32:54.852958+00:00 [error] supervisor: 'esockd_connection_sup - <0.1768.0>', errorContext: connection_crashed, reason: {{case_clause,error},[{emqx_channel,process_alias,2,[{file,"emqx_channel.erl"},{line,1332}]},{emqx_misc,pipeline,3,[{file,"emqx_misc.erl"},{line,140}]},{emqx_channel,process_publish,2,[{file,"emqx_channel.erl"},{line,516}]},{emqx_connection,with_channel,3,[{file,"emqx_connection.erl"},{line,664}]},{emqx_connection,process_msg,2,[{file,"emqx_connection.erl"},{line,364}]},{emqx_connection,process_msg,2,[{file,"emqx_connection.erl"},{line,370}]},{emqx_connection,handle_recv,3,[{file,"emqx_connection.erl"},{line,328}]},{proc_lib,wake_up,3,[{file,"proc_lib.erl"},{line,236}]}]}, offender: [{pid,<0.2066.0>},{name,connection},{mfargs,{emqx_connection,start_link,[[{deflate_options,[]},{max_conn_rate,1000},{active_n,100},{zone,external},{proxy_address_header,<<>>},{proxy_port_header,<<>>},{supported_subprotocols,[]}]]}}]
```

Missing log file can delay the handling of the issue.

### PR DESCRIPTION
**Pull request by thalesmg**
Fixes #6978 .

### CODE REVIEW

**Comment by zmstone** [comment]
Update appup.

**Comment by zmstone** [comment]
Update CHANGES-4.3.md

**Comment by HJianBo** [comment]
Should we put this fix in 4.3.13? Since 4.3.12 is already being prepared for release

**Comment by zmstone** [comment]
> Should we put this fix in 4.3.13? Since 4.3.12 is already being prepared for release

Agree. Hold on, don’t merge before the release is cut

**Comment by thalesmg** [comment]
Ok, I'll update the appups and the changelog after the release is cut then.

File: src/emqx.app.src [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {application, emqx,
  [{id, "emqx"},
   {description, "EMQ X"},
-  {vsn, "4.3.13"}, % strict semver, bump manually!
+  {vsn, "4.3.14"}, % strict semver, bump manually!
```
Comment (zmstone) [inline src/emqx.app.src:4]: NIT: please help to add a comment above to explain why this version is advancing faster than the release version.
I always had to think twice when reviewing the changes.

**Review by zmstone — APPROVED** [review]

**Comment by zmstone** [comment]
nit: bump emqx_dashboard app vsn.

**Review by zmstone — APPROVED** [review]

**Pull request merged**